core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -17,54 +17,56 @@
 
 package org.apache.spark.h2o.backends.external
 
+import org.apache.spark.h2o.H2OFrame
 import org.apache.spark.h2o.converters.WriteConverterCtx
 import org.apache.spark.h2o.converters.WriteConverterCtxUtils.UploadPlan
 import org.apache.spark.h2o.utils.SupportedTypes._
 import org.apache.spark.h2o.utils.{NodeDesc, ReflectionUtils}
 import org.apache.spark.ml.linalg.{DenseVector, SparseVector, Vector}
 import org.apache.spark.sql.types._
-import water.{ExternalFrameConfirmationException, ExternalFrameUtils, ExternalFrameWriterClient}
+import water.fvec.Frame
+import water._
 
 class ExternalWriteConverterCtx(nodeDesc: NodeDesc, totalNumOfRows: Int, writeTimeout: Int, driverTimeStamp: Short) extends WriteConverterCtx {
 
-  val socketChannel = ExternalFrameUtils.getConnection(nodeDesc.hostname, nodeDesc.port, driverTimeStamp)
-  val externalFrameWriter = new ExternalFrameWriterClient(socketChannel)
+  private val externalFrameWriter = ExternalFrameWriterClient.create(nodeDesc.hostname, nodeDesc.port, driverTimeStamp, writeTimeout)
 
-  /**
-    * This method closes the communication after the chunks have been closed
-    * @throws ExternalFrameConfirmationException in case of confirmation failure.
-    *         This has also effect of Spark stopping the current job and
-    *         rescheduling it
-    */
   override def closeChunks(numRows: Int): Unit = {
-    try{
-      externalFrameWriter.waitUntilAllWritten(writeTimeout)
-    } finally {
-      socketChannel.close()
-    }
+    externalFrameWriter.close()
   }
 
   /**
     * Initialize the communication before the chunks are created
     */
   override def createChunks(keystr: String, expectedTypes: Array[Byte], chunkId: Int, maxVecSizes: Array[Int],
                             sparse: Array[Boolean], vecStartSize: Map[Int, Int]): Unit = {
-    externalFrameWriter.createChunks(keystr, expectedTypes, chunkId, totalNumOfRows, maxVecSizes)
+    externalFrameWriter.createChunk(keystr, expectedTypes, chunkId, totalNumOfRows, maxVecSizes)
```

Comment:
what is a difference between `createChunk` and `createChunks` method?

Reply:
They are the same, we changed the name of the method on the H2O backend and this just just updates the code according to that